### ISSUE TITLE
back has issues on Alpine Linux (musl)

### ISSUE THREAD

**Issue by rmottola** [gnustep/libs-back#44]
everything compiles. but on run, I get:
`
Error (objc-load):Error relocating /System/Library/Bundles/libgnustep-back-030.bundle/libgnustep-back-030: XmuLookupStandardColormap: symbol not found`

**Comment by rmottola** [comment]
Maybe @Strahinja is interested in tracking this.

**Comment by Strahinja** [comment]
I can check this out and report if I find anything.

**Comment by DTW-Thalion** [comment]
This doesn't reproduce on current master. The bundle now links libXmu, so `XmuLookupStandardColormap` resolves at load.

The symbol is only left unresolved when the bundle isn't linked against libXmu, and that happens when configure's `AC_CHECK_LIB(Xmu, main)` fails, since its default action is what adds `-lXmu`. Older autoconf generated that check with a `char main();` declaration conflicting with the conftest's `int main(void)`, which strict compilers reject, so `-lXmu` was dropped. The configure in current master was regenerated with autoconf 2.73, which emits `int main(void){ return main(); }` instead; that links, the check passes, and `-lXmu` is added.

I confirmed the current conftest links on Alpine (musl, gcc), so `ac_cv_lib_Xmu_main=yes` and `GRAPHIC_LIBS` gets `-lXmu`. A shared object using `XmuLookupStandardColormap` linked with `-lXmu` gets a `DT_NEEDED` on libXmu and dlopens cleanly; without it, it fails with exactly the error above. libXmu also pulls in libXext/libXt/libX11, so `XShapeCombineMask` resolves the same way.

So this can be closed as fixed by the configure regeneration. One caveat: the check still probes `main` rather than a real libXmu function, so a future regeneration with a stricter autoconf could bring it back; probing `XmuLookupStandardColormap` instead would make it robust.

cc @rmottola @fredkiefer

**Comment by rmottola** [comment]
Yes, I confirm this issue is fixed, I have current running on Alpine.
